crates/query-engine/translation/src/translation/helpers.rs

```diff
@@ -554,3 +554,30 @@ impl NativeQueries {
         }
     }
 }
+
+/// A newtype wrapper around an ndc-spec type which represents accessing a nested field.
+#[derive(Debug, Clone)]
+pub struct FieldPath(pub Vec<String>);
+
+impl From<&Option<Vec<String>>> for FieldPath {
+    fn from(field_path: &Option<Vec<String>>) -> Self {
+        FieldPath(match field_path {
+            // The option has no logical function other than to avoid breaking changes.
+            None => vec![],
+            Some(vec) => vec.clone(),
```

Comment:
Potentially we could use refcast, but this is probably overkill for what is probably a small amount of fairly short vectors.